buildSrc/src/test/java/org/elasticsearch/gradle/BuildPluginIT.java

```diff
@@ -84,7 +85,16 @@ public void testInsecureIvyRepository() throws IOException {
 
     private void runInsecureArtifactRepositoryTest(final String name, final String url, final List<String> lines) throws IOException {
         final File projectDir = getProjectDir("elasticsearch.build");
-        FileUtils.copyDirectory(projectDir, tmpDir.getRoot(), pathname -> pathname.getPath().contains("/build/") == false);
+        final Path projectDirPath = projectDir.toPath();
+        FileUtils.copyDirectory(projectDir, tmpDir.getRoot(), file -> {
+            final Path relativePath = projectDirPath.relativize(file.toPath());
+            for (int i = 0; i < relativePath.getNameCount(); i++) {
```

Comment:
`Path` is itself an `Iterable` so it might be easier to grok this if we replace it with `for (Path segment : relativePath) { }`.

Reply:
Nice, that is a new one to me.